### ISSUE TITLE
[BUG] fail2ban-server is spawned every minute and does not die

### ISSUE THREAD

**Issue by yogo1212** [docker-mailserver/docker-mailserver#2781]
### Miscellaneous first checks

- [X] I checked that all ports are open and not blocked by my ISP / hosting provider.
- [X] I know that SSL errors are likely the result of a wrong setup on the user side and not caused by DMS itself. I'm confident my setup is correct.

### Affected Component(s)

init system

### What happened and when does this occur?

```Markdown
The `dumb-init` process inside the container spawns a `fail2ban-server` process roughly every minute (40 sec).


    721 root      20   0 1387604  29424      0 S   0.5   1.5   0:23.42 dockerd
    637 root      20   0 1208456  20572      0 S   3.7   1.0   0:33.47 containerd
   1073 root      20   0  712392  11120      0 S   0.0   0.6   0:34.08 containerd-shim
  42762 root      20   0   22784   8728      0 R   0.2   0.4   0:32.25 fail2ban-server
  46146 root      20   0   22784   8728      0 R   0.0   0.4   0:26.63 fail2ban-server
  48657 root      20   0   22784   8728      0 R   0.0   0.4   0:22.64 fail2ban-server
  49743 root      20   0   22784   8728      0 R   0.0   0.4   0:20.92 fail2ban-server
  50477 root      20   0   22784   8728      0 R   0.0   0.4   0:19.78 fail2ban-server
  51211 root      20   0   22784   8728      0 R   0.0   0.4   0:18.66 fail2ban-server
  52304 root      20   0   22784   8728      0 R   0.0   0.4   0:16.98 fail2ban-server
  53643 root      20   0   22784   8728      0 R   0.0   0.4   0:15.03 fail2ban-server
  56069 root      20   0   22784   8728      0 R   0.0   0.4   0:11.45 fail2ban-server
  58561 root      20   0   22784   8728      0 R   0.2   0.4   0:07.90 fail2ban-server
  58759 root      20   0   22784   8728      0 R   0.2   0.4   0:07.62 fail2ban-server
  61536 root      20   0   22784   8728      0 R   0.2   0.4   0:03.40 fail2ban-server
  38134 root      20   0   22784   8724      0 R   0.0   0.4   0:40.41 fail2ban-server
  38640 root      20   0   22784   8724      0 R   0.2   0.4   0:39.48 fail2ban-server
  38820 root      20   0   22784   8724      0 R   0.0   0.4   0:39.16 fail2ban-server
  42583 root      20   0   22784   8724      0 R   0.0   0.4   0:32.54 fail2ban-server
  43319 root      20   0   22784   8724      0 R   0.0   0.4   0:31.33 fail2ban-server
  45434 root      20   0   22784   8724      0 R   0.0   0.4   0:27.80 fail2ban-server
  45620 root      20   0   22784   8724      0 R   0.0   0.4   0:27.49 fail2ban-server
  46500 root      20   0   22784   8724      0 R   0.2   0.4   0:26.06 fail2ban-server
  48274 root      20   0   22784   8724      0 R   0.2   0.4   0:23.21 fail2ban-server
  48476 root      20   0   22784   8724      0 R   0.0   0.4   0:22.92 fail2ban-server
  48829 root      20   0   22784   8724      0 R   0.2   0.4   0:22.34 fail2ban-server
  49017 root      20   0   22784   8724      0 R   0.2   0.4   0:22.05 fail2ban-server
  49558 root      20   0   22784   8724      0 R   0.0   0.4   0:21.21 fail2ban-server
  49932 root      20   0   22784   8724      0 R   0.0   0.4   0:20.64 fail2ban-server
```

```
$ ps -ef | grep fail2ban-server | wc -l
382
```

Their parent is:
`root        1094    1073  0 08:40 ?        00:00:00 /usr/bin/dumb-init -- supervisord -c /etc/supervisor/supervisord.con`
```


### What did you expect to happen?

```Markdown
Well.. One process should be enough, I think :-p
```

### How do we replicate the issue?

```Markdown
1. (Re-)Start docker-mailserver
2. Wait
```

### DMS version

latest/530663efe865 - v1.11.0, I guess

### What operating system is DMS running on?

Linux

### What instruction set architecture is DMS running on?

x86_64 / AMD64

### What container orchestration tool are you using?

Docker Compose

### docker-compose.yml

```
pretty vanilla.

take the env instead:


ENABLE_FAIL2BAN=1
FAIL2BAN_BLOCKTYPE=drop
```

and, yes:
```
    cap_add:
      - NET_ADMIN
```

### Relevant log output

```bash
fail2ban is only mentioned in supervisord-log:


2022-09-19 16:33:16,731 fail2ban                [55801]: ERROR   Could not start server. Maybe an old socket file is still present. Try to remove /var/run/fail2ban/fail2ban.sock. If you used fail2ban-client to start the server, adding the -x option will do it
/usr/local/bin/fail2ban-wrapper.sh: line 27: /var/run/fail2ban/fail2ban.pid: No such file or directory
/usr/local/bin/fail2ban-wrapper.sh: line 27: kill: `': not a pid or valid job spec
2022-09-19 16:33:54,984 fail2ban                [55902]: ERROR   Could not start server. Maybe an old socket file is still present. Try to remove /var/run/fail2ban/fail2ban.sock. If you used fail2ban-client to start the server, adding the -x option will do it
/usr/local/bin/fail2ban-wrapper.sh: line 27: /var/run/fail2ban/fail2ban.pid: No such file or directory
/usr/local/bin/fail2ban-wrapper.sh: line 27: kill: `': not a pid or valid job spec
2022-09-19 16:34:33,467 fail2ban                [56005]: ERROR   Could not start server. Maybe an old socket file is still present. Try to remove /var/run/fail2ban/fail2ban.sock. If you used fail2ban-client to start the server, adding the -x option will do it
/usr/local/bin/fail2ban-wrapper.sh: line 27: /var/run/fail2ban/fail2ban.pid: No such file or directory
/usr/local/bin/fail2ban-wrapper.sh: line 27: kill: `': not a pid or valid job spec
2022-09-19 16:35:12,182 fail2ban                [56108]: ERROR   Could not start server. Maybe an old socket file is still present. Try to remove /var/run/fail2ban/fail2ban.sock. If you used fail2ban-client to start the server, adding the -x option will do it
/usr/local/bin/fail2ban-wrapper.sh: line 27: /var/run/fail2ban/fail2ban.pid: No such file or directory
/usr/local/bin/fail2ban-wrapper.sh: line 27: kill: `': not a pid or valid job spec
2022-09-19 16:35:50,684 fail2ban                [56217]: ERROR   Could not start server. Maybe an old socket file is still present. Try to remove /var/run/fail2ban/fail2ban.sock. If you used fail2ban-client to start the server, adding the -x option will do it
/usr/local/bin/fail2ban-wrapper.sh: line 27: /var/run/fail2ban/fail2ban.pid: No such file or directory
/usr/local/bin/fail2ban-wrapper.sh: line 27: kill: `': not a pid or valid job spec
2022-09-19 16:36:29,838 fail2ban                [56324]: ERROR   Could not start server. Maybe an old socket file is still present. Try to remove /var/run/fail2ban/fail2ban.sock. If you used fail2ban-client to start the server, adding the -x option will do it
/usr/local/bin/fail2ban-wrapper.sh: line 27: /var/run/fail2ban/fail2ban.pid: No such file or directory
/usr/local/bin/fail2ban-wrapper.sh: line 27: kill: `': not a pid or valid job spec
2022-09-19 16:37:08,960 fail2ban                [56432]: ERROR   Could not start server. Maybe an old socket file is still present. Try to remove /var/run/fail2ban/fail2ban.sock. If you used fail2ban-client to start the server, adding the -x option will do it
/usr/local/bin/fail2ban-wrapper.sh: line 27: /var/run/fail2ban/fail2ban.pid: No such file or directory
/usr/local/bin/fail2ban-wrapper.sh: line 27: kill: `': not a pid or valid job spec
2022-09-19 16:37:48,584 fail2ban                [56533]: ERROR   Could not start server. Maybe an old socket file is still present. Try to remove /var/run/fail2ban/fail2ban.sock. If you used fail2ban-client to start the server, adding the -x option will do it
/usr/local/bin/fail2ban-wrapper.sh: line 27: /var/run/fail2ban/fail2ban.pid: No such file or directory
/usr/local/bin/fail2ban-wrapper.sh: line 27: kill: `': not a pid or valid job spec
2022-09-19 16:38:28,261 fail2ban                [56648]: ERROR   Could not start server. Maybe an old socket file is still present. Try to remove /var/run/fail2ban/fail2ban.sock. If you used fail2ban-client to start the server, adding the -x option will do it
/usr/local/bin/fail2ban-wrapper.sh: line 27: /var/run/fail2ban/fail2ban.pid: No such file or directory
/usr/local/bin/fail2ban-wrapper.sh: line 27: kill: `': not a pid or valid job spec
2022-09-19 16:39:08,169 fail2ban                [56751]: ERROR   Could not start server. Maybe an old socket file is still present. Try to remove /var/run/fail2ban/fail2ban.sock. If you used fail2ban-client to start the server, adding the -x option will do it
```

### What level of experience do you have with Docker and mail servers?

- [ ] I am inexperienced with docker
- [X] I am inexperienced with mail servers
- [ ] I am uncomfortable with the CLI

### Code of conduct

- [X] I have read this project's [Code of Conduct](https://github.com/docker-mailserver/docker-mailserver/blob/master/CODE_OF_CONDUCT.md) and I agree
- [X] I have read the [README](https://github.com/docker-mailserver/docker-mailserver/blob/master/README.md) and the [documentation](https://docker-mailserver.github.io/docker-mailserver/edge/) and I searched the [issue tracker](https://github.com/docker-mailserver/docker-mailserver/issues?q=is%3Aissue) but could not find a solution

### Improvements to this form?

```Markdown
Hmm. Dunno. It's a bit funny. I guess that's the way with forms instead of open forms of human communication.
```

**Comment by georglauterbach** [comment]
Alright, thank you for the report :) Just a short explanation in the beginning: `dumb-init` is not actually spawning these processes, but `supervisord` is. Therefore, the SUpervisor logs are filled with errors. The error clearly states that the socket file is already present.

I don't know why the other F2B instances are not awaited, but in your case, this is a good thing as you can clearly see a problem. I suspect that you performed a restart that caused the problem. Please properly destroy the container (`docker-compose down`), wait a bit, and start the container (`docker-compose up`) again.

**Comment by yogo1212** [comment]
@georglauterbach hi and thank you for your reply :-)

as you requested, i ran `docker-compose down` (or rather: systemd stopped the docker-compose service when i rebooted).
after a reboot, systemd ran `docker-compose up` and there is a new container:

```
[root@mail ~]# docker ps
CONTAINER ID   IMAGE                                 COMMAND                  CREATED              STATUS                        PORTS                                                                                                                                                                                                                       NAMES
9e4d45aac83c   mailserver/docker-mailserver:latest   "/usr/bin/dumb-init …"   About a minute ago   Up About a minute (healthy)   0.0.0.0:25->25/tcp, :::25->25/tcp, 0.0.0.0:143->143/tcp, :::143->143/tcp, 0.0.0.0:465->465/tcp, :::465->465/tcp, 0.0.0.0:587->587/tcp, :::587->587/tcp, 110/tcp, 995/tcp, 0.0.0.0:993->993/tcp, :::993->993/tcp, 4190/tcp   mailserver
```

but there are new fail2ban processes:

```
ps -ef | grep fail2ba
root        1934    1093 65 11:28 ?        00:00:47 /usr/bin/python3 /usr/bin/fail2ban-server --async -b -s /var/run/fail2ban/fail2ban.sock -p /var/run/fail2ban/fail2ban.pid --loglevel INFO --logtarget /var/log/fail2ban.log --syslogsocket auto
root        2677    1093 48 11:29 ?        00:00:17 /usr/bin/python3 /usr/bin/fail2ban-server --async -b -s /var/run/fail2ban/fail2ban.sock -p /var/run/fail2ban/fail2ban.pid --loglevel INFO --logtarget /var/log/fail2ban.log --syslogsocket auto
root        2858    1140  0 11:29 ?        00:00:00 /bin/bash /usr/local/bin/fail2ban-wrapper.sh
root        2859    2858  9 11:29 ?        00:00:00 /usr/bin/python3 /usr/bin/fail2ban-client start
root        2860    2859  9 11:29 ?        00:00:00 [fail2ban-server] <defunct>
root        2862    1093  0 11:29 ?        00:00:00 /usr/bin/python3 /usr/bin/fail2ban-server --async -b -s /var/run/fail2ban/fail2ban.sock -p /var/run/fail2ban/fail2ban.pid --loglevel INFO --logtarget /var/log/fail2ban.log --syslogsocket auto
```

i've checked the socket file - it isn't there?!
maybe it's only there for a brief moment - don't know how to check that.
```
root@mail:/var/run/fail2ban# ls -al
total 12
drwxr-xr-x 2 root root 4096 Jun 23 07:44 .
drwxr-xr-x 1 root root 4096 Sep 20 11:36 ..
```

here's from after another restart of the container:
```
root@mail:/var/run/fail2ban# ps -ef | grep fail2b
root         769       6  0 11:36 ?        00:00:00 /bin/bash /usr/local/bin/fail2ban-wrapper.sh
root         771     769  0 11:36 ?        00:00:00 /usr/bin/python3 /usr/bin/fail2ban-client start
root         794     771  0 11:36 ?        00:00:00 [fail2ban-server] <defunct>
root         812       1 76 11:36 ?        00:00:18 /usr/bin/python3 /usr/bin/fail2ban-server --async -b -s /var/run/fail2ban/fail2ban.sock -p /var/run/fail2ban/fail2ban.pid --loglevel INFO --logtarget /var/log/fail2ban.log --syslogsocket auto
root        1518     660  0 11:37 pts/0    00:00:00 grep fail2b
```

note the ppids - one defunct process is spawned by the wrapper script.
all other instances are run with ppid 1 - the `dumb-init` process - which was what led me to believe that `dumb-init` was at fault.
```
root@mail:/var/run/fail2ban# ps -ef | grep fail2b
root         812       1 67 11:36 ?        00:00:44 /usr/bin/python3 /usr/bin/fail2ban-server --async -b -s /var/run/fail2ban/fail2ban.sock -p /var/run/fail2ban/fail2ban.pid --loglevel INFO --logtarget /var/log/fail2ban.log --syslogsocket auto
root        1555       6  0 11:37 ?        00:00:00 /bin/bash /usr/local/bin/fail2ban-wrapper.sh
root        1559       1 49 11:37 ?        00:00:15 /usr/bin/python3 /usr/bin/fail2ban-server --async -b -s /var/run/fail2ban/fail2ban.sock -p /var/run/fail2ban/fail2ban.pid --loglevel INFO --logtarget /var/log/fail2ban.log --syslogsocket auto
root        1644     660  0 11:37 pts/0    00:00:00 grep fail2b
root@mail:/var/run/fail2ban# ps -ef | head       
UID          PID    PPID  C STIME TTY          TIME CMD
root           1       0  0 11:36 ?        00:00:00 /usr/bin/dumb-init -- supervisord -c /etc/supervisor/supervisord.conf
root           6       1  0 11:36 ?        00:00:00 /usr/bin/python3 /usr/bin/supervisord -c /etc/supervisor/supervisord.conf
```

here's from the same boot after a few minutes:
```
root         812       1 38 11:36 ?        00:01:18 /usr/bin/python3 /usr/bin/fail2ban-server --async -b -s /var/run/fail2ban/fail2ban.sock -p /var/run/fail2ban/fail2ban.pid --loglevel INFO --logtarget /var/log/fail2ban.log --syslogsocket auto
root        1559       1 29 11:37 ?        00:00:48 /usr/bin/python3 /usr/bin/fail2ban-server --async -b -s /var/run/fail2ban/fail2ban.sock -p /var/run/fail2ban/fail2ban.pid --loglevel INFO --logtarget /var/log/fail2ban.log --syslogsocket auto
root        1664       1 23 11:38 ?        00:00:31 /usr/bin/python3 /usr/bin/fail2ban-server --async -b -s /var/run/fail2ban/fail2ban.sock -p /var/run/fail2ban/fail2ban.pid --loglevel INFO --logtarget /var/log/fail2ban.log --syslogsocket auto
root        1774       1 20 11:38 ?        00:00:19 /usr/bin/python3 /usr/bin/fail2ban-server --async -b -s /var/run/fail2ban/fail2ban.sock -p /var/run/fail2ban/fail2ban.pid --loglevel INFO --logtarget /var/log/fail2ban.log --syslogsocket auto
root        1872       1 18 11:39 ?        00:00:10 /usr/bin/python3 /usr/bin/fail2ban-server --async -b -s /var/run/fail2ban/fail2ban.sock -p /var/run/fail2ban/fail2ban.pid --loglevel INFO --logtarget /var/log/fail2ban.log --syslogsocket auto
```

**Comment by georglauterbach** [comment]
Ah, interesting. I did not look close enough: the PPID of `/usr/bin/python3 /usr/bin/fail2ban-server ...` is indeed 1, and this is because F2B is running as a daemon. My process tree looks like this:

``` CONSOLE
root@mail:/# ps -ajfx
   PPID     PID    PGID     SID  UID   TIME COMMAND
      0       1       1       1    0   0:00 /usr/bin/dumb-init -- supervisord -c /etc/supervisor/supervisord.conf
      1       7       7       7    0  13:11 /usr/bin/python3 /usr/bin/supervisord -c /etc/supervisor/supervisord.conf
      7       8       8       7    0   0:00  \_ /bin/bash /usr/local/bin/start-mailserver.sh
      8     892       8       7    0   0:00  |   \_ tail -Fn 0 /var/log/mail/mail.log
...
      7     651     651       7  108   0:30  \_ /usr/sbin/opendmarc -f -p inet:8893@localhost -P /var/run/opendmarc/opendmarc.pid
      7     721     721       7    0   4:46  \_ /bin/bash /usr/local/bin/fail2ban-wrapper.sh
    721 1278539     721       7    0   0:00  |   \_ sleep 5
...
      7  556481  556481       7    0   2:37  \_ /bin/bash /usr/local/bin/postfix-wrapper.sh
 556481 1278541  556481       7    0   0:00  |   \_ sleep 5
...
      1     771     769     769    0  92:28 /usr/bin/python3 /usr/bin/fail2ban-server --async -b -s /var/run/fail2ban/fail2ban.sock -p /var/run/fail2ban/fail2ban.pid --loglevel INFO --logt
      1  556479  556479  556479    0   0:07 /usr/lib/postfix/sbin/master
 556479  556487  556479  556479  101   0:02  \_ qmgr -l -t unix -u
 556479  560402  556479  556479  101   0:01  \_ tlsmgr -l -t unix -u -c
 556479 1277137  556479  556479  101   0:00  \_ pickup -l -t fifo -u -c -o content_filter=
```

It shows that all processes running as daemons run with `PPID` 1 which makes sense :D But the process creating new instances is Supervisor nevertheless. Now we need to know why `/var/run/fail2ban/fail2ban.pid` is not created, i.e. why F2B won't start. Please disable F2B manually

``` CONSOLE
root@mail:/# supervisorctl stop fail2ban # we do not want more F2B processes
root@mail:/# killall fail2ban # kill all F2B processes; maybe this command does not work, kill them manually in this case
root@mail:/# /usr/bin/fail2ban-client start
```

Please show us the output of the last command.

**Comment by yogo1212** [comment]
Sure!

```
root@mail:/# /usr/bin/fail2ban-client start
2022-09-21 05:58:26,833 fail2ban                [1584]: ERROR   Could not start server. Maybe an old socket file is still present. Try to remove /var/run/fail2ban/fail2ban.sock. If you used fail2ban-client to start the server, adding the -x option will do it
root@mail:/# ps -ef | grep fail2ban
root        1587       1 98 05:57 ?        00:00:33 /usr/bin/python3 /usr/bin/fail2ban-server --async -b -s /var/run/fail2ban/fail2ban.sock -p /var/run/fail2ban/fail2ban.pid --loglevel INFO --logtarget /var/log/fail2ban.log --syslogsocket auto
root@mail:/var/run/fail2ban# ls -al
total 12
drwxr-xr-x 2 root root 4096 Jun 23 07:44 .
drwxr-xr-x 1 root root 4096 Sep 21 05:57 .. # time not from the previous start
```

**Comment by yogo1212** [comment]
interesting..

```
/usr/bin/python3 /usr/bin/fail2ban-server --async -f -s /var/run/fail2ban/fail2ban.sock -p /var/run/fail2ban/fail2ban.pid --loglevel INFO --logtarget /var/log/fail2ban.log --syslogsocket auto
```

doesn't appear to be doing anything. not even writing to the log file.
changing the `-b` to `-f`, both files are created and the log is written:

```
root@mail:/var/run/fail2ban# ls -al
total 16
drwxr-xr-x 1 root root 4096 Sep 21 06:10 .
drwxr-xr-x 1 root root 4096 Sep 21 05:57 ..
-rw------- 1 root root    5 Sep 21 06:10 fail2ban.pid
srwx------ 1 root root    0 Sep 21 06:10 fail2ban.sock
```

**Comment by yogo1212** [comment]
i've created a world-writable file `/tmp/qwer` but i can't get fail2ban to log to there when using `-b`.
sadly, that means that there's no output from `-v` that might lead to an explanation.
well.. the good thing is that we no know that log files appear to suffer from a similar problem as other file system access.

but is this still a docker-mailserver issue or a fail2ban issue?

```
/usr/bin/python3 /usr/bin/fail2ban-server --async -b -s /var/run/fail2ban/fail2ban.sock -p /var/run/fail2ban/fail2ban.pid -vvv --logtarget /tmp/qwer
 +   44 7F404324B740 fail2ban.configreader     INFO  Loading configs for fail2ban under /etc/fail2ban 
 +   45 7F404324B740 fail2ban.configreader     DEBUG Reading configs for fail2ban under /etc/fail2ban 
 +   46 7F404324B740 fail2ban.configreader     DEBUG Reading config files: /etc/fail2ban/fail2ban.conf
 +   47 7F404324B740 fail2ban.configparserinc  INFO    Loading files: ['/etc/fail2ban/fail2ban.conf']
 +   47 7F404324B740 fail2ban.configparserinc  TRACE     Reading file: /etc/fail2ban/fail2ban.conf
 +   48 7F404324B740 fail2ban.configparserinc  INFO    Loading files: ['/etc/fail2ban/fail2ban.conf']
 +   48 7F404324B740 fail2ban.configparserinc  TRACE     Shared file: /etc/fail2ban/fail2ban.conf
 +   49 7F404324B740 fail2ban                  INFO  Using socket file /var/run/fail2ban/fail2ban.sock
 +   49 7F404324B740 fail2ban                  INFO  Using pid file /var/run/fail2ban/fail2ban.pid, [INFO] logging to /tmp/qwer
 +   50 7F404324B740 fail2ban                  DEBUG   direct starting of server in 6385, deamon: True
 +   64 7F404324B740 fail2ban.server           INFO  Starting in daemon mode
 +   65 7F404324B740 fail2ban                  DEBUG Exit with code 0
 ```

**Comment by georglauterbach** [comment]
To be honest, I think this is neither a DMS nor a F2B issue, but most likely a user-related issue, i.e. a problem related to your specific setup. Unless other users or maintainers can recreate it, we will need to assume the error is on your side. There were no F2B changes in the current `:edge` or `:latest` build.

I do not know why the socket file is not created. Are you using special file systems, volume mounts, etc. anything that could be interesting (and which deviates from a certain "standard" (e.g. you're not using EXT4 but NFS)? Your `docker-compose.yml` is "vanilly`, could you share it with us nevertheless?

**Comment by yogo1212** [comment]
This the template used by Ansible to create the docker-compose file:

```
services:
  mailserver:
    image: docker.io/mailserver/docker-mailserver:latest
    container_name: mailserver
    # If the FQDN for your mail-server is only two labels (eg: example.com),
    # you can assign this entirely to `hostname` and remove `domainname`.
    hostname: mail
    domainname: {{ domain }}
    env_file: mailserver.env
    # To avoid conflicts with yaml base-60 float, DO NOT remove the quotation marks.
    ports:
      - "25:25"    # SMTP  (explicit TLS => STARTTLS)
      - "143:143"  # IMAP4 (explicit TLS => STARTTLS)
      - "465:465"  # ESMTP (implicit TLS)
      - "587:587"  # ESMTP (explicit TLS => STARTTLS)
      - "993:993"  # IMAP4 (implicit TLS)
    volumes:
      - {{ mail_mount }}/data:/var/mail/
      - {{ mail_mount }}/state:/var/mail-state/
      - {{ mail_mount }}/log:/var/log/mail/
      - {{ mail_mount }}/config:/tmp/docker-mailserver/
      - /etc/localtime:/etc/localtime:ro
      - /etc/letsencrypt:/etc/letsencrypt:ro
    restart: always
    stop_grace_period: 1m
    cap_add:
      - NET_ADMIN
    healthcheck:
      test: "ss --listening --tcp | grep -P 'LISTEN.+:smtp' || exit 1"
      timeout: 3s
      retries: 0
```

Running on Rocky Linux 9 and docker from `https://download.docker.com/linux/centos/docker-ce.repo`.

**Comment by yogo1212** [comment]
here's a diff between our mailserver.env and the one in this repo:
```
< OVERRIDE_HOSTNAME=
---
> OVERRIDE_HOSTNAME=mail.{{domain}}
94c94
< SPOOF_PROTECTION=
---
> SPOOF_PROTECTION=1
128c128
< ENABLE_FAIL2BAN=0
---
> ENABLE_FAIL2BAN=1
155c155
< SSL_TYPE=
---
> SSL_TYPE=letsencrypt
187c187
< POSTFIX_MAILBOX_SIZE_LIMIT=
---
> POSTFIX_MAILBOX_SIZE_LIMIT=10737418240 # should be 10 GB
197c197
< POSTFIX_MESSAGE_SIZE_LIMIT=
---
> POSTFIX_MESSAGE_SIZE_LIMIT=15728640 # should be 15 MB
289c289
< ENABLE_SPAMASSASSIN=0
---
> ENABLE_SPAMASSASSIN=1
```

**Comment by yogo1212** [comment]
hmm. as the whole machine is set up using ansible, it might be possible to reproduce the error that way....
let me check tomorrow

**Comment by georglauterbach** [comment]
I rather believe that this has to do with Rocky Linux - I believe we had F2B issues with Rocky Linux in the past. @casperklein Do you have an idea / do you remember the Rocky Linux F2B issues?

**Comment by casperklein** [comment]
I fired up a rocky test VM, installed docker/docker-compose and took the DMS default configuration (with the exception of `ENABLE_FAIL2BAN=1`)

I couldn't reproduce the behaviour yogo1212 reported. But fail2ban seems not to correctly work either:

```
[root@localhost mail]# docker exec -it mailserver setup fail2ban
[  ERROR  ]  Fail2ban not running
[  ERROR  ]  Aborting
```

Currently I don't have much more time to debug. But I would also conclude, that the problem(s) are Rocky Linux specific.

The other Rocky issues from the past where related to podman and iptables/nftables: [Issues](https://github.com/docker-mailserver/docker-mailserver/issues?q=is%3Aissue+sort%3Aupdated-desc+rocky+is%3Aclosed)

**Comment by yogo1212** [comment]
wow!
going through the other issues, i had an idea that might have something to do with namespaces.

indeed, fail2ban closes all file descriptors in `__createDaemon` (server/server.py)

here's where the magic happens:

```python
        with open("/tmp/test.txt", "a") as myfile:
            myfile.write("pre close fd\n")

        try:
            maxfd = os.sysconf("SC_OPEN_MAX")
        except (AttributeError, ValueError):
            maxfd = 256    # default maximum

        # urandom should not be closed in Python 3.4.0. Fixed in 3.4.1
        # http://bugs.python.org/issue21207
        if sys.version_info[0:3] == (3, 4, 0): # pragma: no cover
            urandom_fd = os.open("/dev/urandom", os.O_RDONLY)
            for fd in range(0, maxfd):
                try:
                    if not os.path.sameopenfile(urandom_fd, fd):
                        os.close(fd)
                except OSError:   # ERROR (ignore)
                    pass
            os.close(urandom_fd)
        else:
            os.closerange(0, maxfd)

        with open("/tmp/test.txt", "a") as myfile:
            myfile.write("pre redir stdio\n")
```

only the first line is printed..
perhaps one of those file descriptors belongs to a mount namespace and is keeping it open?
need to do more digging. anyway - this is now definitely a fail2ban issue.

**Comment by polarathene** [comment]
Closing as related to resolved #2792 :+1:
